src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -465,6 +460,72 @@ public static IPackageSearchMetadata FindPackage(
             return packagesList.OrderByDescending(p => p.Identity.Version).FirstOrDefault();
         }
 
+        private static IOrderedEnumerable<IPackageSearchMetadata> ApplyPackageSort(IEnumerable<IPackageSearchMetadata> query, domain.PackageOrder orderBy)
+        {
+            switch (orderBy)
+            {
+                case domain.PackageOrder.LastPublished:
+                    return query
+                        .OrderByDescending(q => q.Published)
+                        .ThenBy(q => q.Identity.Id)
+                        .ThenBy(q => q.Identity.Version);
+
+                case domain.PackageOrder.Id:
+                    return query.OrderBy(q => q.Identity.Id)
+                        .ThenBy(q => q.Title)
+                        .ThenBy(q => q.Identity.Version);
+
+                case domain.PackageOrder.Popularity:
+                    return query
+                        .OrderByDescending(q => q.DownloadCount)
+                        .ThenByDescending(q => q.VersionDownloadCount)
+                        .ThenBy(q => q.Identity.Id);
+
+                case domain.PackageOrder.Title:
+                    return query
+                        .OrderBy(q => q.Title)
+                        .ThenBy(q => q.Identity.Id)
+                        .ThenBy(q => q.Identity.Version);
+
+                default:
+                    // Since we are returning an ordered enumerable, we need
+                    // to apply some kind of order, even if no order has been specified.
+                    return query.OrderBy(_ => 0);
+            }
+        }
+
+        private static SearchOrderBy? GetSortOrder(domain.PackageOrder orderBy, bool useMultiVersionOrdering)
+        {
+            switch (orderBy)
+            {
+                case domain.PackageOrder.Popularity:
+                    if (useMultiVersionOrdering)
+                    {
+                        return SearchOrderBy.DownloadCountAndVersion;
+                    }
+                    else
+                    {
+                        return SearchOrderBy.DownloadCount;
+                    }
+
+                case domain.PackageOrder.Id:
+                case domain.PackageOrder.Title:
+                    return SearchOrderBy.Id;
+
+                default:
+                    if (orderBy != domain.PackageOrder.Unsorted)
+                    {
+                        "chocolatey".Log().Warn("Order By '{0}' has been specified. This order is done on client side and may lead to inconsistently ordered results.", orderBy);

```

Comment:
Can we have some extra words here?  A short reason to explain why this might happen, i.e. due to the paging that is happening on the server, or something else.